javaparser-core/src/main/java/com/github/javaparser/resolution/declarations/ResolvedReferenceTypeDeclaration.java

```diff
@@ -47,6 +47,11 @@ default ResolvedReferenceTypeDeclaration asReferenceType() {
         return this;
     }
 
+    @Override
+    default boolean isReferenceType() {
+        return true;
```

Comment:
I'm not sure a declaration can be considered as a type.

Reply:
At least, the `ResolvedReferenceType` and `ResolvedType` implement the method "asReferenceType".
https://github.com/javaparser/javaparser/blob/dfbd466829fca9047b995b5bf9a28343b81d7c5a/javaparser-core/src/main/java/com/github/javaparser/resolution/declarations/ResolvedReferenceTypeDeclaration.java#L45-L53